cli/src/commands/extensions/api.rs

```diff
@@ -8,38 +8,54 @@ use std::str::FromStr;
 use anyhow::{anyhow, Context, Error, Result};
 use deno_runtime::deno_core::{op, OpDecl, OpState};
 use phylum_types::types::auth::{AccessToken, RefreshToken};
-use phylum_types::types::common::{JobId, ProjectId};
+use phylum_types::types::common::JobId;
 use phylum_types::types::job::JobStatusResponse;
 use phylum_types::types::package::{
     Package, PackageDescriptor, PackageStatusExtended, PackageType,
 };
 use phylum_types::types::project::ProjectDetailsResponse;
+use serde::{Deserialize, Serialize};
 use tokio::fs;
 
 use crate::auth::UserInfo;
 use crate::commands::extensions::state::ExtensionState;
-use crate::commands::parse::{self, get_packages_from_lockfile, LOCKFILE_PARSERS};
+use crate::commands::parse::{self, LOCKFILE_PARSERS};
 use crate::config::get_current_project;
 
+/// Package descriptor for any ecosystem.
+#[derive(Serialize, Deserialize, Debug)]
+struct PackageSpecifier {
+    name: String,
+    version: String,
+}
+
+impl From<PackageDescriptor> for PackageSpecifier {
+    fn from(descriptor: PackageDescriptor) -> Self {
+        Self { name: descriptor.name, version: descriptor.version }
+    }
+}
+
+/// Parsed lockfile content.
+#[derive(Serialize, Deserialize, Debug)]
+struct PackageLock {
+    packages: Vec<PackageSpecifier>,
+    package_type: PackageType,
+}
```

Comment:
I've created custom types here for making our extension API a little easier to use, but all this typesystem repackaging can be somewhat exhausting for our own codebase.

While I don't care that much about the lockfile parsing function just spitting out an array with each package containing the same information over and over again, I think for specifying the packages for analysis this makes things much less confusing to the API consumer. And updating `parseLockfile` just helps with plumbing at that point.